File: src/Servidor/Servicios/Implementacion/DepartamentoServicios.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using NLog;
using Trascend.Bolet.LogicaNegocio.Controladores;
using Trascend.Bolet.ObjetosComunes.ContratosServicios;
using Trascend.Bolet.ObjetosComunes.Entidades;

namespace Trascend.Bolet.Servicios.Implementacion
{
    public class DepartamentoServicios : MarshalByRefObject, IDepartamentoServicios
    {
        private static Logger logger = LogManager.GetCurrentClassLogger();

        /// <summary>
        /// Método que obtiene todos los Departamentos
        /// </summary>
        /// <returns>Todos los departamentos</returns>
        public IList<Departamento> ConsultarTodos()
        {
            #region trace
            if (ConfigurationManager.AppSettings["Ambiente"].ToString().Equals("Desarrollo"))
                logger.Debug("Entrando al Método {0}", (new System.Diagnostics.StackFrame()).GetMethod().Name);
            #endregion

            IList<Departamento> departamentos = ControladorDepartamento.ConsultarTodos();

            #region trace
            if (ConfigurationManager.AppSettings["Ambiente"].ToString().Equals("Desarrollo"))
                logger.Debug("Saliendo del Método {0}", (new System.Diagnostics.StackFrame()).GetMethod().Name);
            #endregion

            return departamentos;
        }


        public Departamento ConsultarPorId(Departamento entidad)
        {
            throw new NotImplementedException();
        }

        public bool InsertarOModificar(Departamento entidad, int hash)
        {
            throw new NotImplementedException();
        }

        public bool Eliminar(Departamento entidad, int hash)
        {
            throw new NotImplementedException();
        }


        public bool VerificarExistencia(Departamento entidad)
        {
            throw new NotImplementedException();
        }
    }
}

File: trunk/src/Servidor/Servicios_Fac/Implementacion/DepartamentoServicios.vb
Imports System.Collections.Generic
Imports System.Configuration
Imports NLog
Imports Diginsoft.Bolet.LogicaNegocio.Controladores
Imports Diginsoft.Bolet.ObjetosComunes.ContratosServicios
Imports Diginsoft.Bolet.ObjetosComunes.Entidades

Namespace Implementacion
    Public Class DepartamentoServicios
        Inherits MarshalByRefObject
        Implements IDepartamentoServicios

        Private Shared logger As Logger = LogManager.GetCurrentClassLogger()


        Public Function ConsultarPorId(ByVal entidad As ObjetosComunes.Entidades.Departamento) As ObjetosComunes.Entidades.Departamento Implements ObjetosComunes.ContratosServicios.IServicioBase(Of ObjetosComunes.Entidades.Departamento).ConsultarPorId
            Throw New NotImplementedException()
        End Function

        Public Function ConsultarTodos() As System.Collections.Generic.IList(Of ObjetosComunes.Entidades.Departamento) Implements ObjetosComunes.ContratosServicios.IServicioBase(Of ObjetosComunes.Entidades.Departamento).ConsultarTodos
            '#Region "trace"
            If ConfigurationManager.AppSettings("Ambiente").ToString().Equals("Desarrollo") Then
                logger.Debug("Entrando al metodo {0}", (New System.Diagnostics.StackFrame()).GetMethod().Name)
            End If
            '#End Region

            Dim departamentos As IList(Of Departamento) = ControladorDepartamento.ConsultarTodos()

            '#Region "trace"
            If ConfigurationManager.AppSettings("Ambiente").ToString().Equals("Desarrollo") Then
                logger.Debug("Saliendo del metodo {0}", (New System.Diagnostics.StackFrame()).GetMethod().Name)
            End If
            '#End Region

            Return departamentos
        End Function

        Public Function Eliminar(ByVal entidad As ObjetosComunes.Entidades.Departamento, ByVal hash As Integer) As Boolean Implements ObjetosComunes.ContratosServicios.IServicioBase(Of ObjetosComunes.Entidades.Departamento).Eliminar
            Throw New NotImplementedException()
        End Function

        Public Function InsertarOModificar(ByVal entidad As ObjetosComunes.Entidades.Departamento, ByVal hash As Integer) As Boolean Implements ObjetosComunes.ContratosServicios.IServicioBase(Of ObjetosComunes.Entidades.Departamento).InsertarOModificar
            Throw New NotImplementedException()
        End Function

        Public Function VerificarExistencia(ByVal entidad As ObjetosComunes.Entidades.Departamento) As Boolean Implements ObjetosComunes.ContratosServicios.IServicioBase(Of ObjetosComunes.Entidades.Departamento).VerificarExistencia
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace